pkg/controller/node/taint_controller.go

```diff
@@ -430,3 +468,12 @@ func (tc *NoExecuteTaintManager) handleNodeUpdate(nodeUpdate *nodeUpdateItem) {
 		tc.processPodOnNode(podNamespacedName, node.Name, tolerations, taints, now)
 	}
 }
+
+func (tc *NoExecuteTaintManager) emitPodDeletionEvent(nsName types.NamespacedName) {
```

Comment:
I think I had also suggested an event when the deletion is canceled (e.g. taint is removed before toleration has expired) -- do you think that would be reasonable?